src/window.rs

```diff
@@ -0,0 +1,149 @@
+//! Displays an image in a window created by sdl2.
+
+use image::RgbaImage;
+use sdl2::event::{WindowEvent, Event};
+use sdl2::rect::Rect;
+use sdl2::keyboard::Keycode;
+use sdl2::pixels::{Color, PixelFormatEnum};
+use sdl2::surface::Surface;
+
+/// Displays the provided RGBA image in a new window.
+pub fn display_image(title: &str, image: &RgbaImage, window_width: u32, window_height: u32) {
+
+    let image_width = image.width();
+    let image_height = image.height();
+
+    // resizes and returns the image that will be used to display in the window
+    let (output_image_width, output_image_height, output_image) = 
+        if image_height < window_height && image_width < window_width {
+            (image_width, image_height, image.clone())
+        } else {
+            // scale is used to determine how small an image has to be resized to fit within
+            // the provided window dimensions
+            let scale = {
```

Comment:
This is wrong if either window_width or window_height is less than 300. You need to define the min window width and height earlier and adjust the user-provided width and height by the time you do this scaling.